### DISCUSSION TITLE
Socket.IO client does not upgrade to WebSocket

### DISCUSSION THREAD

**Discussion by DaRealCodeWritten** [socketio/socket.io#4444] [Q&A]
(Note: I would've started this discussion in the socket.io-client repo but when i try to, it sends me to this one)
I have a Flask webserver using the `flask-socketio` module and `simple-websocket` to drive websocket communication
This is the code im using to communicate with the webserver:
```
function getRosterData() {
    const sock = io("wss://czvr-bot.xyz");
    let uid = create_UUID();
    sock.on("connect", () => {
        console.log("Connected to WSS");
		sock.on("ROSTER REQUEST", (data) => {
            if (data["uuid"] != uid) {

            }
            else {
                let roster = data['data'];
                console.log(roster.toString())
            }
        });
        console.log("Connected to WSS");
        sock.emit("REQUEST ROSTER", {
            "uuid": uid
        });
    });
	return sock;
}
```
(Ignore the weird indents, something went wrong when copy pasting i think)
The client connects to the webserver perfectly fine, however the client appears to make no effort to upgrade to WebSocket communication and sticks to polling as evidenced by:
```
INFO:werkzeug:127.0.0.1 - - [09/Aug/2022 12:14:03] "GET /socket.io/?EIO=4&transport=polling&t=OA41mP2 HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Aug/2022 12:14:03] "POST /socket.io/?EIO=4&transport=polling&t=OA41ptj&sid=eNqXTRx_nWAW0riFAAAA HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Aug/2022 12:14:03] "GET /socket.io/?EIO=4&transport=polling&t=OA41pvX&sid=eNqXTRx_nWAW0riFAAAA HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Aug/2022 12:14:03] "POST /socket.io/?EIO=4&transport=polling&t=OA41pyH&sid=eNqXTRx_nWAW0riFAAAA HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [09/Aug/2022 12:14:03] "GET /socket.io/?EIO=4&transport=polling&t=OA41pyP&sid=eNqXTRx_nWAW0riFAAAA HTTP/1.1" 200 -
```
Is there a way i can manually trigger the upgrade attempt or am i doing something wrong?

**Comment by darrachequesne** [comment]
Hi! Could you please check the response body of the first request?

It should contain something like:

```
0{"sid":"FSDjX-WRwSA4zTZMALqx","upgrades":["websocket"],"pingInterval": 25000,"pingTimeout": 20000}
```

If the `upgrades` array does not contain `websocket` (i.e. the server indicates that it does not allow upgrades to WebSocket), it could explain why the client does not try WebSocket.

Reference: https://socket.io/docs/v4/how-it-works/#handshake

Which version of the client are you using?

**Reply by DaRealCodeWritten** [reply]
I unfortunately don't know how to check the body of the first request
As for the client version, it's: 4.0.1
Some weird behavior i noticed: i tried creating a socket connection manually in the browser console, and it upgraded to WS, its just the JS file that doesnt want to upgrade
EDIT: Could it be that the Socket client isnt able to upgrade because the code in the `sock.on("connect"...` is preventing it from moving forward with upgrading?
